controllers/hyperparametertuningjob/hyperparametertuningjob_controller_test.go

```diff
@@ -1701,34 +666,140 @@ func createHpoJob(withFinalizer bool, k8sName, k8sNamespace, sageMakerName, regi
+	var actual hpojobv1.HyperparameterTuningJob
+	err := k8sClient.Get(context.Background(), types.NamespacedName{
+		Namespace: tuningJob.ObjectMeta.Namespace,
+		Name:      tuningJob.ObjectMeta.Name,
+	}, &actual)
+	Expect(err).To(HaveOccurred())
+	Expect(apierrs.IsNotFound(err)).To(Equal(true))
+}
+
+// Expect the BestTrainingJob to be set.
+func ExpectBestTrainingJobToBe(tuningJob *hpojobv1.HyperparameterTuningJob, job *sagemaker.HyperParameterTrainingJobSummary) {
+	var actual hpojobv1.HyperparameterTuningJob
+	err := k8sClient.Get(context.Background(), types.NamespacedName{
+		Namespace: tuningJob.ObjectMeta.Namespace,
+		Name:      tuningJob.ObjectMeta.Name,
+	}, &actual)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(*actual.Status.BestTrainingJob.TrainingJobName).To(Equal(*job.TrainingJobName))
+}
+
+// Expect the TrainingJobStatusCounters to be converted and match from the initial description.
+func ExpectTrainingJobStatusCountersToBe(tuningJob *hpojobv1.HyperparameterTuningJob, counters *sagemaker.TrainingJobStatusCounters) {
+	var actual hpojobv1.HyperparameterTuningJob
+	err := k8sClient.Get(context.Background(), types.NamespacedName{
+		Namespace: tuningJob.ObjectMeta.Namespace,
+		Name:      tuningJob.ObjectMeta.Name,
+	}, &actual)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(*actual.Status.TrainingJobStatusCounters.Completed).To(Equal(*counters.Completed))
+}
+
+// Helper function to create a DescribeHyperParameterTuningJobOutput.
+func CreateDescribeOutputWithOnlyStatus(status sagemaker.HyperParameterTuningJobStatus, bestTrainingJob *sagemaker.HyperParameterTrainingJobSummary, statusCounters *sagemaker.TrainingJobStatusCounters) sagemaker.DescribeHyperParameterTuningJobOutput {
+	return sagemaker.DescribeHyperParameterTuningJobOutput{
 		HyperParameterTuningJobStatus: status,
+		BestTrainingJob:               bestTrainingJob,
+		TrainingJobStatusCounters:     statusCounters,
 	}
-	err := k8sClient.Status().Update(context.Background(), job)
-	Expect(err).ToNot(HaveOccurred())
+}
+
+// Helper function to verify that the specified object is a StopHyperParameterTuningJobInput and that it requests to delete the HyperParameterTuningJob.
+func ExpectRequestToStopHyperParameterTuningJob(req interface{}, tuningJob *hpojobv1.HyperparameterTuningJob) {
+	Expect(req).To(BeAssignableToTypeOf((*sagemaker.StopHyperParameterTuningJobInput)(nil)))
+
+	stopRequest := req.(*sagemaker.StopHyperParameterTuningJobInput)
+	Expect(*stopRequest.HyperParameterTuningJobName).To(Equal(controllers.GetGeneratedJobName(tuningJob.ObjectMeta.GetUID(), tuningJob.ObjectMeta.GetName(), MaxHyperParameterTuningJobNameLength)))
+}
+
+// Helper function to verify that the controller attempted to delete the spawned training jobs.
+func ExpectDeletedSpawnedTrainingJobs(spawner mockTrackingHPOTrainingJobSpawner, calls int) {
+	Expect(spawner.deleteSpawnedTrainingJobsCalls.Len()).To(Equal(calls))
```

Comment:
We shouldn't reduce the quality of these tests -- the unit tests before checked that we called the spawner deletion correctly. Can you re-add the tests for this, just like you did with `ExpectSpawnMissingTrainingJobs`?